internal/nix/shell.go

```diff
@@ -235,15 +236,44 @@ func (s *Shell) Run(nixShellFilePath string) error {
 		return errors.WithStack(cmd.Run())
 	}
 
-	cmd := exec.Command("nix-shell", "--command", s.execCommand(), "--pure")
-	cmd.Args = append(cmd.Args, toKeepArgs(env, buildAllowList(s.env))...)
-	cmd.Args = append(cmd.Args, nixShellFilePath)
-	cmd.Env = env
+	var cmd *exec.Cmd
+	if featureflag.NixlessShell.Enabled() {
+		// Get the required env vars from nix, and then spawn a shell directly.
+		vars, err := s.computeNixShellEnv(nixShellFilePath)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		shellrc, err := s.writeDevboxShellrc(vars)
+		if err != nil {
+			// We don't have a good fallback here, since all the variables we need for anything to work
+			// are in the shellrc file. For now let's fail. Later on, we should remove the vars from the
+			// shellrc file. That said, one of the variables we have to evaluate ($shellHook), so we need
+			// the shellrc file anyway (unless we remove the hook somehow).
+			debug.Log("Failed to write devbox shellrc: %s", err)
+			return errors.WithStack(err)
+		}
+		// Link other files that affect the shell settings and environments.
+		s.linkShellStartupFiles(filepath.Dir(shellrc))
```

Comment:
ah, this `linkShellStartupFiles` is what I was referring to in demo-day. If you look into the implementation, we have only filled it out for `zsh`. We should probably fill it out for other shells as well. WDYT?

Reply:
ah i see.. well, yeah, it would make sense (but it should be separate from this)